### PR TITLE
Fixed config of ngx_http_concat_module

### PR DESCRIPTION
**Pull request by mathieu-aubin**

### CODE REVIEW

**Comment by mathieu-aubin** [comment]
this also allows for the module to be built dynamically

File: modules/ngx_http_concat_module/config [hunk 1]
```diff
@@ -1,4 +1,20 @@
-ddon_name=ngx_http_concat_module
-HTTP_MODULES="$HTTP_MODULES ngx_http_concat_module"
-HTTP_AUX_FILTER_MODULES="$HTTP_AUX_FILTER_MODULES ngx_http_concat_module"
-NGX_ADDON_SRCS="$NGX_ADDON_SRCS $ngx_addon_dir/ngx_http_concat_module.c"
+ngx_addon_name=ngx_http_concat_module
+
+if test -n "$ngx_module_link"; then
+
+    ngx_module_type=HTTP_FILTER
```
Comment (chobits) [inline modules/ngx_http_concat_module/config]: HTTP_FILTER -> HTTP_AUX_FILTER

Although HTTP_FILTER can work from auto/modules, I think HTTP_AUX_FILTER is more precise.

```
...
    if [ -z "$ngx_module_order" -a \
         \( "$ngx_module_type" = "HTTP_FILTER" \
         -o "$ngx_module_type" = "HTTP_AUX_FILTER" \) ]
    then
        eval ${ngx_module}_ORDER=\"$ngx_module_name \
                                   ngx_http_copy_filter_module\"
    else
        eval ${ngx_module}_ORDER=\"$ngx_module_order\"
    fi
...
```

**Comment by chobits** [comment]
ACK for other lines

**Comment by mathieu-aubin** [comment]
The last requested changes have been tested on
- nginx-1.9.10 (without dynamic modules)
- nginx-1.9.11 (with dynamic modules)

**Comment by chobits** [comment]
Thanks! Merged

**Pull request merged**